panic: vault-env / process signal

**Describe the bug**:
We're trying to run vault-env in daemon mode and it typically fails on the first try but succeeds on the second try. This panic does not happen when you run in a non-daemon mode.

**Expected behaviour**:
A panic gets raised and the pod restarts and comes back up.

**Steps to reproduce the bug**:
Run a Pod with daemon mode enabled.

**Additional context**:
Trying to run vault-env in daemon mode (injector) and it typically fails on the first try but succeeds on the second try, unclear why. This doesn't seem related to a specific service, it's happening everywhere where daemon mode is enabled. This service is running inside Istio.

**Environment details**:
- Kubernetes version (e.g. v1.10.2): v1.14.9-eks
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): EKS
- bank-vaults version (e.g. 0.4.17): 1.0.0
- Install method (e.g. helm or static manifests): helm
- Logs from the misbehaving component (and any other relevant logs):

```
time="2020-04-06T19:13:12Z" level=info msg="received new Vault token"
time="2020-04-06T19:13:12Z" level=info msg="initial Vault token arrived"
time="2020-04-06T19:13:12Z" level=info msg="spawning process: [server] app=vault-env
time="2020-04-06T19:13:12Z" level=info msg="in daemon mode..." app=vault-env
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4d496b]
goroutine 65 [running]:
os.(*Process).signal(0x0, 0x1502260, 0xc000512008, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:56 +0x3b
os.(*Process).Signal(...)
	/usr/local/go/src/os/exec.go:131
main.main.func2(0xc000290360, 0xc0003e8580, 0x153d120, 0xc0002d79d0)
	/build/cmd/vault-env/main.go:235 +0x77
created by main.main
	/build/cmd/vault-env/main.go:233 +0xdcb
```

/kind bug

--- a/cmd/vault-env/main.go
+++ b/cmd/vault-env/main.go
@@ -230,6 +230,11 @@ func main() {
 
 		signal.Notify(sigs)
 
+		err = cmd.Start()
+		if err != nil {
+			logger.Fatalln("failed to start process", entrypointCmd, err.Error())
+		}
+
 		go func() {
 			sig := <-sigs
 			err := cmd.Process.Signal(sig)
@@ -240,7 +245,7 @@ func main() {
 			}
 		}()
 
-		err = cmd.Run()
+		err = cmd.Wait()
 
 		close(sigs)
 

